x/profiles/types/models_chain_links.go

```diff
@@ -202,6 +204,57 @@ func (b Base58Address) VerifyPubKey(key cryptotypes.PubKey) (bool, error) {
 
 // --------------------------------------------------------------------------------------------------------------------
 
+var _ AddressData = &HexAddress{}
+
+// NewHexAddress returns a new HexAddress instance
+// NOTE: Currently it only supports keccak256-uncompressed addresses
+func NewHexAddress(value string) *HexAddress {
+	return &HexAddress{Value: value}
+}
+
+// Validate implements AddressData
+func (h HexAddress) Validate() error {
+	if strings.TrimSpace(h.Value) == "" {
+		return fmt.Errorf("value cannot be empty or blank")
+	}
+
+	if len(h.Value) <= 2 {
+		return fmt.Errorf("address cannot be smaller than prefix")
+	}
+
+	prefix, addrWithoutPrefix := h.Value[:2], h.Value[2:]
+	if prefix != "0x" {
+		return fmt.Errorf("address prefix does not match the 0x prefix")
+	}
```

Comment:
Honestly I'm not very convinced of these checks. There might be some chains that do not have any prefix before the real address. What we should do instead is handling the `0x` prefix as something that is completely optional IMO

Reply:
Since there might be different prefix from `0x`, should we add prefix into `HexAddress`? Then, we can skip prefix checking like:
```
_, addrWithoutPrefix := h.Value[:len(h.Prefix)], h.Value[len(h.Prefix):]
```

Reply:
@dadamu Before deciding to implement this back, do you have a handful of examples of chains that use hex-encoded addresses, and the prefixes they use? It would be better to analyze that before deciding how to do things

Reply:
@RiccardoM Nope, currently only eth applies hex address. Other chains almost use `bech32` or `base58`.

Reply:
@dadamu Ok, thanks. Honestly, since the hex encoding can only have one prefix (which is `0x`), I think we can assume that the `Value` of a `HexAddress` can have or not such prefix. For this reason, I think we can implement the `Prefix` field as you did previously, allowing it to also be empty in the case that a chain does not use it. The only check we will have to do then is to make sure that the `Value` begins with the prefix.